go/private/rules/cgo.bzl

```diff
@@ -107,29 +130,35 @@ def cgo_configure(go, srcs, cdeps, cppopts, copts, cxxopts, clinkopts):
                 _include_unique(cppopts, "-iquote", inc, seen_quote_includes)
             for inc in cc_system_includes(d):
                 _include_unique(cppopts, "-isystem", inc, seen_system_includes)
-            for lib in cc_libs(d):
-                # If both static and dynamic variants are available, Bazel will only give
-                # us the static variant. We'll get one file for each transitive dependency,
-                # so the same file may appear more than once.
-                if (lib.basename.startswith("lib") and
-                    has_simple_shared_lib_extension(lib.basename)):
-                    # If the loader would be able to find the library using rpaths,
-                    # use -L and -l instead of hard coding the path to the library in
-                    # the binary. This gives users more flexibility. The linker will add
-                    # rpaths later. We can't add them here because they are relative to
-                    # the binary location, and we don't know where that is.
-                    libname = lib.basename[len("lib"):lib.basename.rindex(".")]
-                    clinkopts.extend(["-L", lib.dirname, "-l", libname])
-                    inputs_direct.append(lib)
-                elif (lib.basename.startswith("lib") and
-                      has_versioned_shared_lib_extension(lib.basename)):
-                    # With a versioned shared library, we must use the full filename,
-                    # otherwise the library will not be found by the linker.
-                    libname = ":%s" % lib.basename
-                    clinkopts.extend(["-L", lib.dirname, "-l", libname])
-                    inputs_direct.append(lib)
+            for lib in as_iterable(d[CcInfo].linking_context.libraries_to_link):
+                static_lib = get_static_lib(lib)
+                if static_lib == None and lib.dynamic_library:
+                    dylib = lib.dynamic_library
+                    # If both static and dynamic variants are available, Bazel will only give
+                    # us the static variant. We'll get one file for each transitive dependency,
+                    # so the same file may appear more than once.
+                    if has_simple_shared_lib_extension(dylib.basename):
```

Comment:
Don't we still need to check for the `lib` prefix? We remove it unconditionally below.